### PR TITLE
Added Imported Symbols, exported symbols and opcode ngrams

### PR DESCRIPTION
**Pull request by Sidu28**
Added some alphabetical features - imported symbols, exported symbols as well as ngrams from opcodes.  For now, the code prints out the first 10 ngrams.  
One thing that I didn't have the time to fix was that pprint no longer seems to work?  I'm not 100% sure why.  For this commit, I have used a simple print.

### CODE REVIEW

File: feature_utils.py [hunk 1]
```diff
@@ -16,8 +16,9 @@
 }
 
 ALPHABETICAL_FEATURE_EXTRACTORS = {
-  features.imported_symbols.ImportedSymbolsExtractor: None
-
+  features.imported_symbols.ImportedSymbolsExtractor: None,
+  features.exported_symbols.ExportedSymbolsExtractor: None,
+  features.assembly_ngrams.AssembleNgrams: None
```
Comment (mananshah99) [inline feature_utils.py:21]: Seems like the `AssemblyNgrams` extractor should be a part of the ASMExtractor.

File: features/asm.py [hunk 2]
```diff
@@ -15,13 +15,13 @@ def disassemble(self):
...
       for j in md.disasm(section.get_data(), self.pefile_parsed.OPTIONAL_HEADER.ImageBase + section.VirtualAddress):
         disassembly += "0x%x:\t%s\t%s\n" % (j.address, j.mnemonic, j.op_str)
-    
     return disassembly
   
   def extract(self, **kwargs):
     features = {
       'eax_count': None
     }
+
```
Comment (mananshah99) [inline features/asm.py:24]: Would likely make more sense to put the ngrams logic here and return that as a part of the extract function

File: features/assembly_ngrams.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+        disassembly = []
+        for section in self.pefile_parsed.sections:
+            md = Cs(CS_ARCH_X86, CS_MODE_32)
+            for j in md.disasm(section.get_data(),
+                               self.pefile_parsed.OPTIONAL_HEADER.ImageBase + section.VirtualAddress):
+                disassembly.append(j.mnemonic)
+
+        disassembly = self.generate_ngrams(disassembly)
+        return disassembly
```
Comment (mananshah99) [inline features/assembly_ngrams.py:29]: The above comments would ensure that this code is not repeated

File: features/exported_symbols.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        self.pefile_parse()
+        pe = self.pefile_parsed
+
+        export_list = []
+
+        if hasattr(pe, 'DIRECTORY_ENTRY_EXPORT'):
+            for exp in pe.DIRECTORY_ENTRY_EXPORT.symbols:
+                export_list.append(exp.name)
+
+            features['export_list'] = export_list
```
Comment (mananshah99) [inline features/exported_symbols.py:21]: Might be better to make these features binary, having them alphabetical does not make much sense

File: features/imported_symbols.py [hunk 5]
```diff
@@ -16,13 +16,14 @@ def extract(self, **kwargs):
...
-        features['import_list'] = import_list
+        if hasattr(pe, 'DIRECTORY_ENTRY_IMPORT'):
+            for entry in pe.DIRECTORY_ENTRY_IMPORT:
+                DLL = entry.dll
+                for imp in entry.imports:
+                    address = hex(imp.address)
+                    import_name = imp.name
+                    import_list.append(import_name)
+    
+            features['import_list'] = import_list
```
Comment (mananshah99) [inline features/imported_symbols.py:27]: Same here, binary features would make more sense (does a pe file have import X? 0/1) as opposed to the import names

**Review by mananshah99 — CHANGES_REQUESTED** [review]

File: features/exported_symbols.py (hunk 4 above)
Comment (Sidu28) [reply features/exported_symbols.py:21]: Not quite sure what you mean here? I'm trying to extract the export symbol names.  I believe those are the most relevant given what I have read.

File: features/imported_symbols.py (hunk 5 above)
Comment (Sidu28) [reply features/imported_symbols.py:27]: Ok, but how would that work for our features?  I have 0's for all import I am not making and 1's for the few things that I do import?

**Comment by mananshah99** [comment]
Yeah, seems like that would be more conducive to ML than the strings, which aren’t super informative as is and not strictly necessary for a language model.

File: features/exported_symbols.py (hunk 4 above)
Comment (mananshah99) [reply features/exported_symbols.py:21]: Same as the below discussion

File: features/imported_symbols.py (hunk 5 above)
Comment (mananshah99) [reply features/imported_symbols.py:27]: Answered in a comment below

**Comment by Sidu28** [comment]
Ok, i'll make each import an individual feature.

**Comment by Sidu28** [comment]
@mananshah99 Code has been updated, please review and approve if ok

**Review by mananshah99 — APPROVED** [review]
LGTM, thanks for the changes!

**Pull request merged**